File: examples/eventemitter.pl
use v5.14;

use Mojo::EventEmitter;


package Cat {
    use Mojo::Base 'Mojo::EventEmitter';
    sub poke {
        my ($self, $times) = @_;
        $self->emit(roar => $times);
    }
    sub kill {
        my ($self, $times) = @_;
        $self->emit('dead');
    }
}


my $tiger = Cat->new();

$tiger->on(roar => sub {
    my ($self, $times) = @_;
    for (1 .. $times) {
        say 'RAWR!';
    }
});

$tiger->once(dead => sub {
    say '(x.x)';
});

$tiger->poke(2);
$tiger->poke(2);
$tiger->kill();
$tiger->kill();

File: t/examples/eventemitter.py
import Pyjo.Test


class NoseTest(Pyjo.Test.NoseTest):
    script = __file__
    srcdir = '../..'


class UnitTest(Pyjo.Test.UnitTest):
    script = __file__


if __name__ == '__main__':

    from Pyjo.Test import *  # noqa

    import Pyjo.EventEmitter

    plan_tests(5)

    class Cat(Pyjo.EventEmitter.object):
        def poke(self, times):
            self.emit('roar', times)

        def kill(self):
            self.emit('dead')

    tiger = Cat()

    @tiger.on
    def roar(cat, times):
        for _ in range(0, times):
            pass_ok('RAWR!')

    @tiger.once
    def dead(cat):
        pass_ok('(x.x)')

    tiger.poke(2)
    tiger.poke(2)
    tiger.kill()
    tiger.kill()

    done_testing()
